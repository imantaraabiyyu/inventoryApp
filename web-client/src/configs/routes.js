import { HomePage } from "../scenes/home";
import { ItemsPage, ItemPage } from "../scenes/item";
import { StocksPage, StockPage } from "../scenes/stock";
import { TransactionsPage, TransactionPage } from "../scenes/transaction";
import { UnitsPage, UnitPage } from "../scenes/unit";
import { ErrorPage } from "../scenes/error";

const routes = [
  {
    path: "/",
    breadcrumb: "Home",
    component: HomePage,
    exact: true
  },
  {
    path: "/items/add",
    breadcrumb: "Add Item",
    component: ItemPage
  },
  {
    path: "/items/:id",
    breadcrumb: "Detail",
    component: ItemPage
  },
  {
    path: "/items",
    breadcrumb: "Items",
    component: ItemsPage
  },
  {
    path: "/stocks/add",
    breadcrumb: "Add Stock",
    component: StockPage
  },
  {
    path: "/stocks/:id",
    breadcrumb: "Detail",
    component: StockPage
  },
  {
    path: "/stocks",
    breadcrumb: "Stocks",
    component: StocksPage
  },
  {
    path: "/transactions/add",
    breadcrumb: "Add Transaction",
    component: TransactionPage
  },
  {
    path: "/transactions/:id",
    breadcrumb: "Detail",
    component: TransactionPage
  },
  {
    path: "/transactions",
    breadcrumb: "Transactions",
    component: TransactionsPage
  },
  {
    path: "/units/add",
    breadcrumb: "Add Unit",
    component: UnitPage
  },
  {
    path: "/units/:id",
    breadcrumb: "Detail",
    component: UnitPage
  },
  {
    path: "/units",
    breadcrumb: "Units",
    component: UnitsPage
  },
  {
    path: "*",
    component: ErrorPage,
    props: {
      code: 404
    }
  }
];

export default routes;
