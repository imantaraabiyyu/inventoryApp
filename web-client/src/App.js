import React, { Component } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import routes from "./configs/routes";
import "./App.css";
import Page from "./components/Page";

class App extends Component {
  render() {
    return (
      <Router basename="/webclient">
        <Page>
          <Switch>
            {routes.map((route, index) => (
              <Route
                key={index}
                path={route.path}
                exact={route.exact}
                render={(props) => (
                  <route.component {...props} {...route.props} />
                )}
              />
            ))}
          </Switch>
        </Page>
      </Router>
    );
  }
}

export default App;
