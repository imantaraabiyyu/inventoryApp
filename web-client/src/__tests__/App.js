import React from "react";
import { shallow } from "enzyme";
import App from "../App";
import routes from "../configs/routes";

describe("App", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
    expect(component).toMatchSnapshot();
  });

  it("should have N route(s)", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find("Route")).toHaveLength(routes.length);
  });
});
