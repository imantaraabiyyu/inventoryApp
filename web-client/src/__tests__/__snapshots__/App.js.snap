// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<BrowserRouter>
  <div>
    <h2>
      Welcome to React
    </h2>
  </div>
  <Switch>
    <Route
      exact={true}
      key="0"
      path="/"
      render={[Function]}
    />
    <Route
      key="1"
      path="/items/add"
      render={[Function]}
    />
    <Route
      key="2"
      path="/items/:id"
      render={[Function]}
    />
    <Route
      key="3"
      path="/items"
      render={[Function]}
    />
    <Route
      key="4"
      path="/stocks"
      render={[Function]}
    />
    <Route
      key="5"
      path="/transactions"
      render={[Function]}
    />
    <Route
      key="6"
      path="/units"
      render={[Function]}
    />
    <Route
      key="7"
      path="*"
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
